src/tiered/tiered_handle.c

```diff
@@ -26,6 +26,9 @@ __tiered_dhandle_setup(WT_SESSION_IMPL *session, WT_TIERED *tiered, uint32_t i,
             id = WT_TIERED_INDEX_LOCAL;
         else if (type == WT_DHANDLE_TYPE_TIERED)
             id = WT_TIERED_INDEX_LOCAL;
+        else if (type == WT_DHANDLE_TYPE_TIERED_TREE)
+            /* Nothing to do for this type. */
+            goto err;
```

Comment:
Not sure if you want to add this in as a comment, but this should be removed as a TODO in WT-7538.